File: 260324/잔해물을 덮기 위한 사각형의 최소 넓이/minimum-area-of-rectangle-to-cover-debris.py

# Please write your code here.
OFFSET = 10
MAX = 2 * OFFSET
blocks = [[0] * (MAX + 1) for _ in range(MAX + 1)]

for n in range(2):
    x1, y1, x2, y2 = map(int, input().split())
    x1, y1, x2, y2 = x1 + OFFSET, y1 + OFFSET, x2 + OFFSET, y2 + OFFSET
    for i in range(x1, x2):
        for j in range(y1, y2):
            blocks[i][j] = n + 1

area = 0
min_x, min_y = 2 * OFFSET, 2 * OFFSET
max_x, max_y = -2 * OFFSET, -2 * OFFSET

for i in range(MAX + 1):
    for j in range(MAX + 1):
        if blocks[i][j] == 1:
            min_x = min(min_x, i)
            min_y = min(min_y, j)
            max_x = max(max_x, i)
            max_y = max(max_y, j)

size = (max_x - min_x + 1) * (max_y - min_y + 1)
print(size)

File: code_tree/trail2/잔해물을 덮기 위한 사각형의 최소 넓이/minimum-area-of-rectangle-to-cover-debris.cpp
#include <iostream>
using namespace std;

#define OFFSET 1000
#define MAX_N 2001

int arr[MAX_N][MAX_N] = {};

int main() {
    int x1, y1, x2, y2;


    cin >> x1 >> y1 >> x2 >> y2;

    x1 += OFFSET;
    y1 += OFFSET;
    x2 += OFFSET;
    y2 += OFFSET;

    for (int i = x1; i < x2; i++) {
        for (int j = y1; j < y2; j++) {
            arr[i][j] = 1;
        }
    }


    int a1, b1, a2, b2;
    cin >> a1 >> b1 >> a2 >> b2;

    a1 += OFFSET;
    b1 += OFFSET;
    a2 += OFFSET;
    b2 += OFFSET;

    for (int i = a1; i < a2; i++) {
        for (int j = b1; j < b2; j++) {
            arr[i][j] = 0;
        }
    }
    int min_x = MAX_N, min_y = MAX_N;
    int max_x = 0, max_y = 0;

    for (int i = 0; i < MAX_N; i++) {
        for (int j = 0; j < MAX_N; j++) {
            if (arr[i][j] == 1) {
                if (min_x > i) min_x = i;
                if (min_y > j) min_y = j;
                if (max_x < i) max_x = i;
                if (max_y < j) max_y = j;
            }
        }
    }

    int size = (max_x - min_x + 1) * (max_y - min_y + 1);

    if (size >= 4000000)
        cout << 0 << endl;
    else 
        cout << size << endl;

    return 0;
}